cpp/src/arrow/chunked_array.cc

```diff
@@ -55,7 +55,7 @@ ChunkedArray::ChunkedArray(ArrayVector chunks, std::shared_ptr<DataType> type)
         << "cannot construct ChunkedArray from empty vector and omitted type";
     type_ = chunks_[0]->type();
   }
-
+  ARROW_CHECK_LE(chunks.size(), std::numeric_limits<int>::max());
```

Comment:
Is this limit useful if it ends up not making performance better anyway?

Reply:
This is useful in ensuring we are not generating more chunks than can be addressed by the index type.